library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -0,0 +1,652 @@
+        outputPort.emit((T)tuple);
+      }
+    }
+  }
+
+  @Override
+  public void endWindow()
+  {
+    try {
+      if (currentWindowId > windowManager.getLargestRecoveryWindow()) {
+        if (currentWindowRecoveryState == null) {
+          currentWindowRecoveryState = new MutablePair<String, String>();
+        }
+        if (lower != null && upper != null) {
+          previousUpperBound = upper;
+          isPolled = true;
+        }
+        MutablePair<String, String> windowBoundaryPair = new MutablePair<>(lower, upper);
+        currentWindowRecoveryState = windowBoundaryPair;
+        windowManager.save(currentWindowRecoveryState, operatorId, currentWindowId);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("saving recovery", e);
+    }
+    currentWindowRecoveryState = new MutablePair<>();
+  }
+
+  public int getPartitionCount()
+  {
+    return partitionCount;
+  }
+
+  public void setPartitionCount(int partitionCount)
+  {
+    this.partitionCount = partitionCount;
+  }
+
+  @Override
+  public void activate(Context cntxt)
+  {
+    if (context.getValue(OperatorContext.ACTIVATION_WINDOW_ID) != Stateless.WINDOW_ID
+        && context.getValue(OperatorContext.ACTIVATION_WINDOW_ID) < windowManager.getLargestRecoveryWindow()) {
+      // If it is a replay state, don't start any threads here
+      return;
+    }
+  }
+
+  @Override
+  public void deactivate()
+  {
+    try {
+      if (dbPoller != null && dbPoller.isAlive()) {
+        dbPoller.interrupt();
+        dbPoller.join();
+      }
+    } catch (InterruptedException ex) {
+      // log and ignore, ending execution anyway
+      LOG.error("exception in poller thread: ", ex);
+    }
+  }
+
+  @Override
+  public void handleIdleTime()
+  {
+    if (execute) {
+      try {
+        Thread.sleep(spinMillis);
+      } catch (InterruptedException ie) {
+        throw new RuntimeException(ie);
+      }
+    } else {
+      LOG.error("Exception: ", cause);
+      DTThrowable.rethrow(cause.get());
+    }
+  }
+
+  protected void replay(long windowId) throws SQLException
+  {
+    isReplayed = true;
+
+    MutablePair<String, String> recoveredData = new MutablePair<String, String>();
+    try {
+      recoveredData = (MutablePair<String, String>)windowManager.load(operatorId, windowId);
+
+      if (recoveredData != null) {
+        //skip the window and return if there was no incoming data in the window
+        if (recoveredData.left == null || recoveredData.right == null) {
+          return;
+        }
+
+        if (recoveredData.right.equals(rangeQueryPair.getValue()) || recoveredData.right.equals(previousUpperBound)) {
+          LOG.info("Matched so returning");
+          return;
+        }
+
+        JdbcPollInputOperator jdbcPoller = new JdbcPollInputOperator();
+        jdbcPoller.setStore(store);
+        jdbcPoller.setKey(getKey());
+        jdbcPoller.setPartitionCount(getPartitionCount());
+        jdbcPoller.setPollInterval(50);
```

Comment:
Should this be a constant? If so, can you declare this as a _DEFAULT at the top?